numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -30,23 +30,54 @@ class FakeShape(tuple):
     indexing, similar to the shape in CUDA Python. (Numpy shape arrays allow
     negative indexing)
     '''
+
     def __getitem__(self, k):
         if isinstance(k, int) and k < 0:
             raise IndexError('tuple index out of range')
         return super(FakeShape, self).__getitem__(k)
 
 
+class FakeKernelCUDAArray(object):
+    def __init__(self, item):
+        assert isinstance(item, FakeCUDAArray)
+        self.__dict__['_item'] = item
+
+    def __wrap_if_fake(self, item):
```

Comment:
A minor point, but it seems strange that this method begins with a double underscore but isn't a dunder method.

Reply:
This is probably deliberate - it results in a mangled name of `_FakeKernelCUDAArray__wrap_if_fake`, which ensures it does not conflict with any attributes of `self._item`

Reply:
yup